spraycharles.py

```diff
@@ -277,68 +588,97 @@ def main(passlist, userlist, host, module, path, csvfile, attempts, interval, eq
         new_users = check_file_contents(userfile, users)
         new_passwords = check_file_contents(passfile, passwords)
-        
+
         if len(new_users) > 0:
-            colors.color_print(f'[>] Adding {len(new_users)} new users into the spray!', colors.blue)
+            console.print(
+                f"[>] Adding {len(new_users)} new users into the spray!", style="info"
+            )
             users.extend(new_users)
 
         if len(new_passwords) > 0:
-            colors.color_print(f'[>] Adding {len(new_passwords)} new passwords to the end of the spray!', colors.blue)
+            console.print(
+                f"[>] Adding {len(new_passwords)} new passwords to the end of the spray!",
+                style="info",
+            )
             passwords.extend(new_passwords)
 
         # print line separator
         if len(new_passwords) > 0 or len(new_users) > 0:
             print()
 
-        for username in users:
-            if domain:
-                username = f'{domain}\\{username}'
-            if jitter is not None:
-                if jitter_min is None:
-                    jitter_min = 0
-                time.sleep(random.randint(jitter_min,jitter))
-            login(target, username, password, csvfile)
-            
+        with Progress(transient=True) as progress:
+            task = progress.add_task(f"[green]Spraying: {password}", total=len(users))
+            while not progress.finished:
+                for username in users:
+                    if domain:
+                        username = f"{domain}\\{username}"
+                    if jitter is not None:
+                        if jitter_min is None:
+                            jitter_min = 0
+                        time.sleep(random.randint(jitter_min, jitter))
+                    login(target, username, password, csvfile)
+                    progress.update(task, advance=1)
+
             # log the login attempt
-            logging.info(f'Login attempted as {username}')
-            
+            logging.info(f"Login attempted as {username}")
+
         login_attempts += 1
-    
+
     # analyze the results to point out possible hits
     analyzer = analyze.Analyzer(csvfile, notify, webhook, host, total_hits)
-    success = analyzer.analyze()
+    new_hit_total = analyzer.analyze()
 
-    # Pausing if specified by user before continuing with spray
-    if success is True and pause:
+    if new_hit_total > total_hits and pause:
         print()
-        colors.color_print('[+] Successful login potentially identified. Pausing...', colors.yellow)
+        console.print(
+            "[+] Successful login potentially identified. Pausing!",
+            style="bold bright_green",
+        )
         print()
-        input('Press enter to continue:')
+        Confirm.ask(
+            "[blue]Press enter to continue",
+            default=True,
+            show_choices=False,
+            show_default=False,
+        )
```

Comment:
I might be misinterpreting this pause-related block, but I believe this is triggering only after the spray is completely done. When this code hits and pauses, if you press enter the script just exits. This block can probably get deleted since it should be hitting in the `check_sleep()` call right?

Reply:
Okay this makes sense. I was always a tad confused about what was happening here. Will pull it.